app/src/main/java/com/example/habithelper/utilities/LinearRegressionCalculator.java

```diff
@@ -21,13 +22,18 @@ public LinearRegressionCalculator() {
      * @return an array of coefficients corresponding to the correlation of each habit to the user's mood
      */
     public void performLeastSquares(List<TrackDay> daysTracked, int numDaysTracked, int numHabits) {
-        moodList = new double[numDaysTracked];
-        habitHistoryList = new double[numDaysTracked][numHabits];
+        moodArray = new double[numDaysTracked];
+        habitHistoryArray = new double[numDaysTracked][numHabits];
 
         putParseDataIntoLists(daysTracked, numDaysTracked);
 
         HabitMultipleLinearRegression habitMLR = new HabitMultipleLinearRegression();
-        habitMLR.newSampleData(moodList, habitHistoryList);
+        try {
+            habitMLR.loadAndCheckData(moodArray, habitHistoryArray);
+        }
+        catch(Exception e) {
+            Log.e("LinearRegressionCalculator", "data does not fit parameters for MLR");
```

Comment:
Can we scope the exception that does not fit the parameters for MLR. Right now this will catch any exception type